infra/modules/providers/azure/service-principal/main.tf

```diff
@@ -28,4 +28,8 @@ resource "azuread_service_principal_password" "sp" {
   service_principal_id = azuread_service_principal.sp[0].object_id
   value                = local.sp_password
   end_date_relative    = var.sp_pwd_end_date_relative
+
+  lifecycle {
```

Comment:
These lifecycle hooks bake in the original passwords inserted into the creation of the service principal password and the key vault secrets that store them. How will password updates be managed due to this change? Outside of terraform?

Reply:
Yes should be completed outside of TF